### ISSUE TITLE
timestamps for number

### ISSUE THREAD

**Issue by Atefeh197** [m-bain/whisperX#314]
Hi everyone

I just use whisperX, it has more accurate timestamps than whisper, but it is very inaccurate in the case of numbers. 

For example, the text for the first row is "Good day." and it extracts accurate timestamps for each word.
But the second row is "780 802", you see the start and end times are very close to each other furthermore we do not have time for "780" and "802" separately.

{'start': 2.861, 'end': 4.081, 'text': 'Good day.', 'words': [{'word': 'Good', 'start': 2.861, 'end': 3.281, 'score': 0.587}, {'word': 'day.', 'start': 3.301, 'end': 4.001, 'score': 0.48}]},

{'start': 20.025, 'end': 20.045, 'text': '780 802', 'words': [{'word': '780'}, {'word': '802'}]},

How I can get the better timestamps for numbers?

**Comment by m-bain** [comment]
yes, this is a limitation of whisperx, it is unable to provide word timestamps for numerals. You can avoid this by transcribing with `--supress_numerals` flag, this will transcribe numbers literally e.g. "780" -> "seven hundred and eighty". You could then use a text normalizer to convert this back to text

**Comment by Atefeh197** [comment]
Thank you.

I use whisperX like the below:

model.transcribe(audio, batch_size=batch_size)

I must use "--supress_numerals" in this line?

**Comment by Atefeh197** [comment]
when I use 
model = whisperx.load_model("medium", device, compute_type=compute_type, asr_options = {"suppress_numerals" : True} )

it has error
AttributeError: 'str' object has no attribute 'removeprefix'

Do you know about that?

**Comment by sorgfresser** [comment]
Regarding the normalization afterwards, there are libraries like `text2num` but they don't support many languages. Maybe yours is supported.

**Comment by snoop2head** [comment]
@robvanson I think the issue is related to #717

**Comment by snoop2head** [comment]
I found the method on the issue #629  and code snippet below.
https://github.com/m-bain/whisperX/blob/78dcfaab51005aa703ee21375f81ed31bc248560/whisperx/asr.py#L259

Thanks!

**Comment by robvanson** [comment]
@snoop2head 
 "--suppress_numerals" works for me, thanks a lot.

**Comment by villesau** [comment]
Would there be a possibility to implement something that transforms numerics to text where it is needed and reverts back to numeric when the timestamp is set? This way we would get the original numeric value with timestamps in place.

**Comment by Barabazs** [comment]
Should be fixed with #986
